src/Microsoft.Azure.WebJobs.Logging/Internal/LogWriter.cs

```diff
@@ -152,6 +156,15 @@ public Task AddAsync(FunctionInstanceLogItem item, CancellationToken cancellatio
             // Both Start and Completed log here. Completed will overwrite a Start entry. 
             lock (_lock)
             {
+                // Permanent failures when flushing to storage can result in many log entries being buffered.
+                // This basically becomes a memory leak. Mitigate by enforcing a max.
+                if (_activeFuncs.Count >= MaxBufferedEntryCount || _completedFunctions.Count >= MaxBufferedEntryCount)
+                {
+                    _onException?.Invoke(new Exception($"The limit on the number of bufferable log entries was reached. A total of '{MaxBufferedEntryCount}' log entries were dropped."));
```

Comment:
nit: is 'bufferable' a word? :-) Maybe 'queued'?

Reply:
Also -- that total number dropped may not be accurate. Can you use the actual counts for the message?

Reply:
Fixed the non-word :)

See my comment below about how a log entry is often written to both collections. If I show in the error message the active+completed count it will probably be misleading. I'll keep this part as is.